lib/pdf_fill/extras_generator_v2.rb

```diff
@@ -369,37 +370,43 @@ def add_text(value, metadata)
       metadata[:format_options] ||= {}
       metadata[:format_options][:label_width] ||= @default_label_width
 
-      value = apply_humanization(value, metadata[:format_options])
-
-      question_num = metadata[:question_num]
-      if @questions[question_num].blank?
-        question_text = @question_key[question_num]
-        @questions[question_num] = get_question(question_text, metadata)
+      config = find_question_config(metadata)
+      if config && @questions[config[:question_number]].blank?
+        @questions[config[:question_number]] = get_question(config[:question_text], metadata, config)
       end
-
-      @questions[question_num].add_text(value, metadata)
+      value = apply_humanization(value, metadata[:format_options])
+      @questions[config ? config[:question_number] : metadata[:question_num]]&.add_text(value, metadata)
```

Comment:
This method reads really awkwardly, I think because there's all this `config ? config[:question_number] : metadata[:question_num]` logic appearing multiple times, and not always in the same form.

I would recommend doing a few things to make it read better:
- Return early if no config is found (which should never happen anyway). No config means we won't know how to display it at all.
- Settle on a canonical way to determine `question_num`, and always use `@questions[question_num]`.